packages/jsii-spec/lib/spec.ts

```diff
@@ -390,7 +399,7 @@ export interface CollectionTypeReference extends TypeReferenceBase {
         /**
          * The type of an element (map keys are always strings).
          */
-        elementtype: TypeReference;
+        elementtype: TypeInstance<TypeReference>;
```

Comment:
It does not make sense for either array or map elements to be of type `Promise<T>`, or optional.

Reply:
So an array may never contain `null`? Or it'll contain "surprise" `null`s?

Reply:
But that is something else though! Feels like now we're expressing nullability, not optionality.